### DISCUSSION TITLE
Define Click arguments and options from annotation

### DISCUSSION THREAD

**Discussion by cool-RR** [pallets/click#2471] [Q&A] [answered]
Hi,

Is there any package that allows defining Click arguments and options from annotations?

I was imagining usage like this: 

```
@click.command()
def foo(*, bar: Option(int, default=7), is_good: Option(bool)) -> None:
    ...
```

It'll be nice to not have to write decorators in addition to the function signature. I wonder if anyone already thought something like that and implemented it.

**Comment by davidism** [comment, marked as answer]
https://typer.tiangolo.com/

**Comment by cool-RR** [comment]
Awesome, thank you!
